requests/util.js

```diff
@@ -0,0 +1,63 @@
+function createCustomElement(domObjectMap) {
+  const el = document.createElement(domObjectMap.tagName);
+  for (const [key, value] of Object.entries(domObjectMap)) {
+    if (key === 'tagName') {
+      continue;
+    }
+    if (key === 'eventListeners') {
+      value.forEach((obj) => {
+        el.addEventListener(obj.event, obj.func);
+      });
+    }
+    if (key === 'class') {
+      if (Array.isArray(value)) {
+        el.classList.add(...value);
+      } else {
+        el.classList.add(value);
+      }
+    } else if (key === 'child') {
+      el.append(...value);
+    } else {
+      el[key] = value;
+    }
+  }
+  return el;
+}
+
+function getOooQueryParamsString(query) {
+  let queryParam = 'type=OOO';
+  if (
+    query.state !== undefined &&
+    query.state !== null &&
+    query.state !== 'ALL'
+  ) {
+    queryParam += `&state=${query.state}`;
+  }
+  return '?' + queryParam.toString();
+}
```

Comment:
`getOooQueryParamsString`function constructs a query string for OOO requests based on the provided query object.

Effectively handles scenarios where the `query.state` is provided and not `ALL`.

 You can consider using template literals for constructing the query string to improve readability, especially for multiple parameters.

Reply:
used templated literals in a given function. Please check.